### ISSUE TITLE
Estimate costs

### ISSUE THREAD

**Issue by he00741098** [he00741098/pogoot#64]
Rough estimate of costs for current setup

- [ ] $15 per month for AWS (EC2, 3 servers of varying size, medium, small, very small)
- [ ] $0 per month for cloudflare pages and workers (100k requests per day for workers)
- [ ] $0 per month for R2 storage (10gb, 1 million writes, 10 million reads)
- [ ] $0 per month for Turso (1 billion row reads, 25 million row writes, 9gb)

Possible future expansion
- vector database for semantic search
- Hetzner server instead of AWS

Current estimate for serving
- Assuming that the servers can handle every single request within reasonable amounts of time, the limiting factor will be R2.
- Turso can handle 1 billion row reads which is about 33 million reads per day, assuming each request takes less than 11 reads, it can handle at least 3 million requests per day (assuming computation time is unlimited). 
- Turso can handle 25 million row writes which is about 800k writes per day.
- Cloudflare workers can handle 100k requests perday. Assuming that each request takes only 1 cloudflare worker request, 100k notecard sets can be loaded
- Cloudflare R2 stores 10gb, which should be plenty, approximately 10 million notecard sets based on rough calculations. The 1 million writes equates to 33k writes per day, meaning 33 thousand notecards can be stored. The 10 million reads equates to 330k notecards can be fetched per day, which is greater than the amount of pages that cloudflare workers will load for free.
- In conclusion, 33k notecard set writes per day and display 100k notecard sets per day. This is much more than the expected amount of activity the expected amount of users will generate per day. Assuming that a monthly active user writes 3 sets per day and reads 10 sets per day, we can handle about 10k monthly active users.

Minimizing server side rendering would likely lower costs, but that will be difficult.

**Comment by he00741098** [comment]
Closing for now
